File: flang-classic/test/f90_correct/src/check.f90
!
! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
! See https://llvm.org/LICENSE.txt for license information.
! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
!

	subroutine check(result, expect, np)
	integer np
	integer result(np)
	integer expect(np)
	integer tests_passed
	integer tests_failed
	tests_passed = 0
	tests_failed = 0
	do i = 1,np
	    if (expect(i) .ne. result(i)) then
		tests_failed = tests_failed + 1
		print 1,i,result(i),result(i),expect(i),expect(i)
1	format(' --- test number ',i5,' FAILED. res',i5,'(',z8.8,')',
     &		'  exp',i5,'(',z8.8,')')
	    else
		tests_passed = tests_passed + 1
	    endif
	enddo
	if (tests_failed .eq. 0) then
	    print 2,np,tests_passed,tests_failed
	else
	    print 3,np,tests_passed,tests_failed
	endif
2	format(' --- ',i3,' tests completed.',i5,' tests PASSED.',
     &		i5,' tests failed.')
3	format(' --- ',i3,' tests completed.',i5,' tests passed.',
     &		i5,' tests FAILED.')
	end
	subroutine checkd(result, expect, np)
	integer np
	real*8 result(np)
	real*8 expect(np)
	integer tests_passed
	integer tests_failed
	tests_passed = 0
	tests_failed = 0
	do i = 1,np
	    if (expect(i) .ne. result(i)) then
		tests_failed = tests_failed + 1
		print 1,i,result(i),expect(i)
1	format(' --- test number ',i5,' FAILED. res',g12.8,
     &			'  exp',g12.8)
	    else
		tests_passed = tests_passed + 1
	    endif
	enddo
	if (tests_failed .eq. 0) then
	    print 2,np,tests_passed,tests_failed
	else
	    print 3,np,tests_passed,tests_failed
	endif
2	format(' --- ',i3,' tests completed.',i5,' tests PASSED.',
     &		i5,' tests failed.')
3	format(' --- ',i3,' tests completed.',i5,' tests passed.',
     &		i5,' tests FAILED.')
	end

File: flang-classic/test/gnu_extensions/src/check.c
/*
 * Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
 * See https://llvm.org/LICENSE.txt for license information.
 * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 */

#include <stdio.h>
#include <stdlib.h>
#include <assert.h>

extern int __hpf_lcpu;

void
check_(int* res, int* exp, int* np)
{
    int i;
    int n = *np;
    int tests_passed = 0;
    int tests_failed = 0;

    for (i = 0; i < n; i++) {
        if (exp[i] == res[i]) {
	    tests_passed ++;
        } else {
            tests_failed ++;
	    if( tests_failed < 100 )
            printf(
	    "test number %d FAILED. res %d(%08x)  exp %d(%08x)\n",
	     i+1,res[i], res[i], exp[i], exp[i] );
        }
    }
    if (tests_failed == 0) {
	    printf(
	"%3d tests completed. %d tests PASSED. %d tests failed.\n",
                      n, tests_passed, tests_failed);
    } else {
	printf("%3d tests completed. %d tests passed. %d tests FAILED.\n",
                      n, tests_passed, tests_failed);
    }
}

void
check(int* res, int* exp, int* np)
{
    check_(res, exp, np);
}

#if defined(WINNT) || defined(WIN32)
void
__stdcall CHECK(int* res, int* exp, int* np)
{
    check_(res, exp, np);
}
#endif
